lib/merkle-tree.js

```diff
@@ -469,16 +469,22 @@ module.exports = class MerkleTree {
   }
 
   _flushNodes () {
-    // TODO: write neighbors together etc etc
     // TODO: bench loading a full disk page and copy to that instead
     return new Promise((resolve, reject) => {
       const slab = b4a.allocUnsafe(40 * this.flushing.size)
 
       let error = null
-      let missing = this.flushing.size + 1
+      let missing = 1
       let offset = 0
+      let ridx = -1
+
+      const indexes = Array.from(this.flushing.keys()).sort((x, y) => x - y)
+      const rngs = []
```

Comment:
Probably better to avoid abbreviations, `rngs` -> `ranges`, and below there are also two `rng` -> `range`

Btw really awesome improvement

Reply:
@LuKks sounds good, fixed

Reply:
Should I change `rindex` to `rangeIndex`?

Reply:
`rindex` is used very simply so it's ok